src/Portfolio.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.8.0) (token/ERC20/ERC20.sol)
+
+pragma solidity ^0.8.0;
+
+import "lib/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+import "lib/openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+import "lib/chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import "./BasketHandler.sol";
+import "./PriceConverter.sol";
+import "./Coin.sol";
+
+contract Portfolio {
+    function uniswapV3(
+        uint256 amountIn,
+        address tokenIn,
+        address tokenOut
+    ) internal returns (uint256 amountOut) {
+        address router = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+        uint24 fee = 3000;
+        address WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
```

Comment:
These addresses should be passed through the constructor so that they can be network agnostic.

Reply:
Agreed.